### PR TITLE
[WIP] New recipe: Ghostscript

### PR DESCRIPTION
**Pull request by asinghvi17**

### CODE REVIEW

File: G/Ghostscript/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+# install to prefixes
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+platforms = supported_platforms()
+
+products = []
```
Comment (giordano) [inline G/Ghostscript/build_tarballs.jl:35]:
```suggestion
products = Product[]
```

File: G/Ghostscript/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    ),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+# initial setup
+cd $WORKSPACE/srcdir/ghostpdl*
+
+# configure the Makefiles
+./configure --prefix=$prefix --host=${target}
```
Comment (giordano) [inline G/Ghostscript/build_tarballs.jl:22]: We need to generate the `configure` script
```suggestion
./autogen.sh --prefix=${prefix} --host=${target}
```

File: G/Ghostscript/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    ),
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+# initial setup
+cd $WORKSPACE/srcdir/ghostpdl*
+
+# configure the Makefiles
+./autogen.sh --prefix=${prefix} --host=${target}
```
Comment (giordano) [inline G/Ghostscript/build_tarballs.jl:22]:
```suggestion
./autogen.sh --prefix=${prefix} --build=${MACHTYPE} --host=${target}
```

**Comment by giordano** [comment]
Having to wrestle with `autogen.sh` is one of the reasons why I personally don't like using `GitSource`.

**Comment by asinghvi17** [comment]
I suspect the reason Windows fails is because:
> The configure build can be used to build Ghostscript on MSys/Mingw systems, but with a caveat. The msys-dvlpr adds header files into the compiler's header search paths which cause a clash, and the build will fail as a result. If you have the msys-dvlpr package installed, and until a better solution is available you can work around this by temporarily renaming the "\mingw\msys\1.0\include" directory so those headers are no longer found by the compiler.

from https://www.ghostscript.com/doc/9.52/Make.htm#Microsoft_build_64

Is there any way we could do that temporarily during the build?

**Comment by giordano** [comment]
I may be wrong, but I don't think we have the package they mention.  There are [some patches](https://github.com/msys2/MINGW-packages/tree/6e4bb05c00eec5c47f86f07ddefaf4e793a38845/mingw-w64-ghostscript) in the msys2/MINGW repo, but they are for an older version and also quite long, that's why I hadn't spent much time so far looking into them.

**Comment by giordano** [comment]
msys2/MINGW now has [patches](https://github.com/msys2/MINGW-packages/tree/ae59480519593433039dcc2b3ce8e1bf3d3f8ea5/mingw-w64-ghostscript) for Ghostscript 9.53.3, it may be interesting to try to look into them

**Comment by Krastanov** [comment]
I tried to update this pull request in https://github.com/JuliaPackaging/Yggdrasil/pull/2653 (it does not currently acknowledge the work by @asinghvi17 as I do not really know what the way to do that is). The powerpc-linux build now works (probably just a coincidence), but even though the mingw patches are now included, there seems to be other issues with the windows builds.

**Comment by giordano** [comment]
Superseded by #2653

**Pull request closed without merging**